iridium-client-nextjs/src/lib/services/identity.service.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import { OauthConstants } from './oauth-constants';
+import { CookieService } from './cookie.service';
+
+export class IdentityService {
+
+    private cookieService: CookieService;
+    constructor() {
+        this.cookieService = new CookieService();
+    }
+
+    public get(token: string) {
+        useEffect(() => {
+            fetch(process.env.NEXT_PUBLIC_IRIDIUM_DOMAIN + 'identities', {headers: {
+                    'Accept': 'application/vnd.iridium.id.identity-response.1+json', 'Authorization': 'Bearer ' + token,
+                }, method: 'POST'})
+                .then((response) => response.json())
+                .then((res) => {
+                    if (res.status === 200) {
+                        this.cookieService.setCookie('iridium-token', res.access_token, 1, OauthConstants.COOKIE_PATH)
+                    } else {
+                      console.error('error!')
+                    }
+                })
+
+        }, [])
```

Comment:
I have major questions about this piece of code. Is it working as expected? 

.Then chaining in a useEffect is not fun. Wrap it in an IIFE and async/await

`useEffect(() => {
  (async () => {
    const response = await fetch(
      process.env.NEXT_PUBLIC_IRIDIUM_DOMAIN + 'identities',
      {
        headers: {
          Accept: 'application/vnd.iridium.id.identity-response.1+json',
          Authorization: `Bearer ${token}`,
        },
        method: 'POST',
      }
    );
    const res = await response.json();

    if (res.status === 200) {
      this.cookieService.setCookie(
        'iridium-token',
        res.access_token,
        1,
        OauthConstants.COOKIE_PATH
      );
    } else {
      console.error('error!');
    }
  })();
}, []);`

Reply:
Good call.  I meant to remove this.  Thank you.

Reply:
And no, it wasn't working 🙃